lib/glific/templates/interactive_templates.ex

```diff
@@ -384,4 +386,313 @@ defmodule Glific.Templates.InteractiveTemplates do
+
+    Enum.reduce(active_languages, %{}, fn {lang_name, lang_code}, acc ->
+      translations =
+        if lang_name == "English" do
+          {:ok, content_to_translate}
+        else
+          GoogleTranslate.translate(content_to_translate, "English", lang_name,
+            org_id: organization_id
+          )
+        end
+
+      case translations do
+        {:ok, translated_content} ->
+          [title, body | remaining_translations] = translated_content
+
+          global_buttons_length = length(content["globalButtons"])
+
+          global_buttons_translations =
+            Enum.slice(remaining_translations, 0, global_buttons_length)
+
+          items_translations = Enum.drop(remaining_translations, global_buttons_length)
+
+          global_buttons_translated =
+            translate_global_buttons(global_buttons_translations, content["globalButtons"])
+
+          items_translated = do_items_translated(content, items_translations)
+
+          translated_template = %{
+            "title" => title,
+            "body" => body,
+            "globalButtons" => global_buttons_translated,
+            "items" => items_translated,
+            "type" => "list"
+          }
+
+          Map.put(
+            acc,
+            Integer.to_string(Map.get(language_code_map, lang_code)),
+            translated_template
+          )
+      end
+    end)
+  end
+
+  @spec translate_location_request(map(), map(), map(), non_neg_integer()) :: map()
+  defp translate_location_request(content, active_languages, language_code_map, organization_id) do
+    content_to_translate = [content["body"]["text"]]
+
+    Enum.reduce(active_languages, %{}, fn {lang_name, lang_code}, acc ->
+      translations =
+        if lang_name == "English" do
+          {:ok, content_to_translate}
+        else
+          GoogleTranslate.translate(content_to_translate, "English", lang_name,
+            org_id: organization_id
+          )
+        end
+
+      case translations do
+        {:ok, [translated_text]} ->
+          translated_template = %{
+            "action" => content["action"],
+            "body" => %{"text" => translated_text, "type" => content["body"]["type"]},
+            "type" => content["type"]
+          }
+
+          Map.put(
+            acc,
+            Integer.to_string(Map.get(language_code_map, lang_code)),
+            translated_template
+          )
+      end
+    end)
+  end
+
+  @spec build_content_to_translate(map()) :: list
+  defp build_content_to_translate(content) do
+    [
+      content["title"],
+      content["body"]
+    ] ++
+      Enum.map(content["globalButtons"], fn button ->
+        button["title"]
+      end) ++
+      Enum.flat_map(content["items"], fn item ->
+        [
+          item["title"],
+          item["subtitle"]
+        ] ++
+          Enum.flat_map(item["options"], fn option ->
+            [option["title"], option["description"]]
+          end)
+      end)
```

Comment:
save them in different variable and then add them at last to improve readability